src/sqlfluff/api/simple.py

```diff
@@ -1,11 +1,49 @@
 """The simple public API methods."""
 
 from typing import Any, Dict, List, Optional
-from sqlfluff.core import Linter
-from sqlfluff.core import SQLBaseError
+from sqlfluff.core import (
+    dialect_selector,
+    FluffConfig,
+    Linter,
+    SQLBaseError,
+    SQLFluffUserError,
+)
 from sqlfluff.core.linter import ParsedString
 
 
+def get_simple_config(
+    dialect: str = "ansi",
+    rules: Optional[List[str]] = None,
+    exclude_rules: Optional[List[str]] = None,
+    config_path: Optional[str] = None,
+) -> FluffConfig:
+    """Get a config object from simple API arguments."""
+    # Check the requested dialect exists and is valid.
+    try:
+        dialect_selector(dialect)
+    except SQLFluffUserError as err:
+        raise SQLFluffUserError(f"Error loading dialect '{dialect}': {str(err)}")
+    except KeyError:
+        raise SQLFluffUserError(f"Error: Unknown dialect '{dialect}'")
```

Comment:
Suggest you just Pragma no cover these:

```suggestion
    except SQLFluffUserError as err:  # pragma: no cover
        raise SQLFluffUserError(f"Error loading dialect '{dialect}': {str(err)}")
    except KeyError:  # pragma: no cover
        raise SQLFluffUserError(f"Error: Unknown dialect '{dialect}'")
```

Reply:
thanks I've made a test for the second error, but will use pragma for the first one as that's not easy to find a test for 😄 

I'm just investigating whether we can remove those config.py lines at this point 👍

Reply:
I wonder if passing an incorrect file location will trigger the first?

But I’m absolutely fine with pragma no covers for exceptions. They are, by definition, for unexpected cases. Of course if it’s easy to add a case then do, but not essential.

But I just like to keep the 100% cover for a few reasons:
1. Makes it obvious when you do miss cover you should have!
2. Removes noise when % rounding errors look like changes that aren’t cause total number of lines change but number of not cover lines don’t.
3. I’m a perfectionist! 😀

But the pragma no cover is fine to use to achieve that if it’s the right call rather than killing yourself making a complex test case that adds more complexity than real benefit.